HOWTORELEASE.rst

```diff
@@ -10,40 +10,38 @@ taking a lot of time to make a new one.
     pytest releases must be prepared on **Linux** because the docs and examples expect
     to be executed on that platform.
 
-#. Create a branch ``release-X.Y.Z`` with the version for the release.
+The release a version ``MAJOR.MINOR.PATCH``, follow these steps:
 
-   * **maintenance releases**: from ``4.6-maintenance``;
+#. For major and minor releases, create a new branch ``MAJOR.MINOR.x`` from the
+   latest ``master`` and push it to the ``pytest-dev/pytest`` repo.
 
-   * **patch releases**: from the latest ``master``;
+#. Create a branch ``release-MAJOR.MINOR.PATCH`` from the ``MAJOR.MINOR.x`` branch.
+   (Note: historically, the ``4.6.x`` branch is called ``4.6-maintenance``).
 
-   * **minor releases**: from the latest ``features``; then merge with the latest ``master``;
-
-   Ensure your are in a clean work tree.
+   Ensure your are updated and in a clean working tree.
 
 #. Using ``tox``, generate docs, changelog, announcements::
 
-    $ tox -e release -- <VERSION>
+    $ tox -e release -- MAJOR.MINOR.PATCH
 
    This will generate a commit with all the changes ready for pushing.
 
-#. Open a PR for this branch targeting ``master`` (or ``4.6-maintenance`` for
-   maintenance releases).
+#. Open a PR for the ``release-MAJOR.MINOR.PATCH`` branch targeting ``MAJOR.MINOR.x``.
 
 #. After all tests pass and the PR has been approved, publish to PyPI by pushing the tag::
 
-     git tag <VERSION>
-     git push git@github.com:pytest-dev/pytest.git <VERSION>
+     git tag MAJOR.MINOR.PATCH
+     git push git@github.com:pytest-dev/pytest.git MAJOR.MINOR.PATCH
 
    Wait for the deploy to complete, then make sure it is `available on PyPI <https://pypi.org/project/pytest>`_.
 
 #. Merge the PR.
 
-#. If this is a maintenance release, cherry-pick the CHANGELOG / announce
-   files to the ``master`` branch::
+#. Cherry-pick the CHANGELOG / announce files to the ``master`` branch::
 
        git fetch --all --prune
-       git checkout origin/master -b cherry-pick-maintenance-release
-       git cherry-pick --no-commit -m1 origin/4.6-maintenance
+       git checkout origin/master -b cherry-pick-release
+       git cherry-pick --no-commit -m1 origin/MAJOR.MINOR.x
        git checkout origin/master -- changelog
        git commit  # no arguments
```

Comment:
One thing that is not clear for me from the docs: do we keep `MAJOR.MINOR.x` branches around forever, or do we delete them when they go unsupported?

Currently a version goes unsupported when the next minor/major is released: so for example the `5.3` series gets unsupported after `5.4` gets its first release.

I think we should delete `5.3.x` when `5.4.0` is released for the first time. This avoids generating confusion that a bunch of `5.3.x`, `5.4.x`, `5.5.x`, etc, on the repository means that those versions are still supported.

Reply:
My inclination is to keep the branches around forever, for the following reasons, however I don't feel strongly about this. If you prefer to delete, I can update the text - let me know.

1. It makes it easier to switch over to old reference points in git and in GitHub UI. Tags are less convenient because you need to check which is the latest first, while with the branch you just have `.x`. And in GitHub UI switching to old tags is takes more steps than switching to a branch. I do it often in django at least.

2. You never know when you might need that branch anyway, even if it's unsupported.

3. I don't like deleting things :)

Also, I don't *really* think that people will construe the existence of a branch to mean it's still supported, though maybe I'll be proven wrong about it... Confusion/fat finger might happen though.

Reply:
> If you prefer to delete, I can update the text - let me know.

I don't feel strongly about it either, let's see if others chime in. If nobody does, let's go with keeping the branches around given you are the author of the PR. 👍

Reply:
If it becomes a problem we can always start deleting then!